lib/chef/provider/package/cab.rb

```diff
@@ -0,0 +1,150 @@
+          Chef::Log.debug("#{@new_resource} searching for installed package #{package['name']}")
+          found_packages = installed_packages.select { |p| p["package_identity"] =~ /^#{package['name']}~/ }
+          if found_packages.length == 0
+            nil
+          elsif found_packages.length == 1
+            stdout = dism_command("/Get-PackageInfo /PackageName:\"#{found_packages.first["package_identity"]}\"").stdout
+            find_version(stdout)
+          else
+            # Presuming this won't happen, otherwise we need to handle it
+            raise Chef::Exceptions::Package, "Found mutliple packages installed matching name #{package['name']}, found: #{found_packages.length} matches"
+          end
+        end
+
+        def package_version
+          Chef::Log.debug("#{@new_resource} getting product version for package at #{@new_resource.source}")
+          stdout = dism_command("/Get-PackageInfo /PackagePath:\"#{@new_resource.source}\"").stdout
+          find_version(stdout)
+        end
+
+        def find_version(stdout)
+          package_info = parse_dism_get_package_info(stdout)
+          package = split_package_identity(package_info["package_information"]["package_identity"])
+          package["version"]
+        end
+
+        # returns a hash of package state information given the output of dism /get-packages
+        # expected keys: package_identity
+        def parse_dism_get_packages(text)
+          packages = Array.new
+          text.each_line do |line|
+            key, value = line.split(":") if line.start_with?("Package Identity")
+            unless key.nil? || value.nil?
+              package = Hash.new
+              package[key.downcase.strip.tr(" ", "_")] = value.strip.chomp
+              packages << package
+            end
+          end
+          packages
+        end
+
+        # returns a hash of package information given the output of dism /get-packageinfo
+        def parse_dism_get_package_info(text)
+          package_data = Hash.new
+          errors = Array.new
+          in_section = false
+          section_headers = [ "Package information", "Custom Properties", "Features" ]
+          text.each_line do |line|
+            if line =~ /Error: (.*)/
+              errors << $1
+            elsif section_headers.any? { |header| line =~ /^(#{header})/ }
+              in_section = $1.downcase.tr(" ", "_")
+            elsif line =~ /(.*) ?: (.*)/
+              v = $2 # has to be first or the gsub below replaces this variable
+              k = $1.downcase.strip.tr(" ", "_")
+              if in_section
+                package_data[in_section] = Hash.new unless package_data[in_section]
+                package_data[in_section][k] = v
+              else
+                package_data[k] = v
+              end
+            end
+          end
+          unless errors.empty?
+            if !(errors & ["0x80070003\r", "0x80070002\r"]).empty?
```

Comment:
@btm errors is a Array containing error codes, I am verifying here if any one of the code are present in 'errors' since both are raising similar kind of errors. With include? I can verify only one code at a time.  I want if any one of the error code present in the 'errors' it should raise 'file or path not found error' which I am not able to achieve with include? method.